README.md

```diff
@@ -123,34 +123,24 @@ export let loader: LoaderFunction = async ({ request }) => {
 };
 ```
 
-You could also do the `json` and `cors` call in one line.
+If you want to setup it globally once, you can do it like this in `entry.server`
 
 ```ts
-export let loader: LoaderFunction = async ({ request }) => {
-  let data = await getData(request);
-  return await cors(request, json<LoaderData>(data));
+export let handleDataRequest: HandleDataRequestFunction = async (
+  response,
+  { request }
+) => {
+  return await cors(request, response);
 };
 ```
 
-And because `cors` mutates the response, you can also call it and later return.
+Keep in mind you still need to update each response like this:
 
 ```ts
-export let loader: LoaderFunction = async ({ request }) => {
+export let loader: LoaderFunction = async (args) => {
   let data = await getData(request);
   let response = json<LoaderData>(data);
-  await cors(request, response); // this mutates the Response object
-  return response; // so you can return it here
-};
-```
-
-If you want to setup it globally once, you can do it like this in `entry.server`
-
-```ts
-export let handleDataRequest: HandleDataRequestFunction = async (
-  response,
-  { request }
-) => {
-  return await cors(request, response);
+  return handleDataRequest(response, args);
```

Comment:
You shouldn't have to import `handleDataRequest` in your own code, this `handleDataRequest` function is used by Remix after a data request (a request with `?_data` on the URL.